lib/gems/pending/appliance_console/date_time_configuration.rb

```diff
@@ -12,10 +12,10 @@ class DateTimeConfiguration
 
     include ApplianceConsole::Logging
 
-    def initialize
-      @new_date         = nil
-      @new_time         = nil
-      @manual_time_sync = false
+    def initialize(config = {})
+      @new_date         = config.has_key(:date) ? config[:date] : nil
```

Comment:
Also if a Hash, config[:date] is nil if the key doesn't exist, so there's no need for the key check at all given your default...